annotation/src/main/java/org/cbioportal/genome_nexus/annotation/service/internal/VEPVariantAnnotationService.java

```diff
@@ -93,15 +102,16 @@ public VariantAnnotation getAnnotation(String variant)
                 variantAnnotation = this.mapAnnotationJson(variant, annotationJSON);
 
                 // save everything to the cache as a properly parsed JSON
-
-                try {
-                    variantAnnotationRepository.saveAnnotationJson(variant, annotationJSON);
-                }
-                catch (DataIntegrityViolationException e) {
-                    // in case of data integrity violation exception, do not bloat the logs
-                    // this is thrown when the annotationJSON can't be stored by mongo
-                    // due to the variant annotation key being too large to index
-                    LOG.info(e.getLocalizedMessage());
+                if (storeVariantJson) {
+                    try {
+                        variantAnnotationRepository.saveAnnotationJson(variant, annotationJSON);
+                    }
+                    catch (DataIntegrityViolationException e) {
```

Comment:
Can we move this `catch` block outside, I mean as another `catch` statement for the outer `try`? By this way we can avoid a nested try statement. So that the code would simply look like this: 

```Java
if (saveAnnotationJson) {
    variantAnnotationRepository.saveAnnotationJson(variant, annotationJSON);
}
```